cmd/kube-controller-manager/app/controllermanager_test.go

```diff
@@ -219,3 +219,29 @@ func TestTaintEvictionControllerGating(t *testing.T) {
 		})
 	}
 }
+
+func TestNoCloudProviderControllerStarted(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	controllerCtx := ControllerContext{
+		Cloud:    nil,
+		LoopMode: IncludeCloudLoops,
+	}
+
```

Comment:
```suggestion
	controllerCtx.ComponentConfig.Generic.Controllers = []string{"*"}
```

This requires to enable all controllers to run, otherwise it does not try to run

@soltysh with this change it fails without the patch

Reply:
Perfect, let's get that added - thank you!